src/Pic.vue

```diff
@@ -17,10 +19,45 @@
         type: String
       },
       height: {
-        type: String
+        type: String,
+        default: ''
       },
       width: {
-        type: String
+        type: String,
+        default: ''
+      }
+    },
+    computed: {
+      hasWidth () {
+        return this.width !== '';
+      },
+      hasHeight () {
+        return this.height !== '';
+      },
+      computedWidth () {
+        if (this.hasWidth) {
+          return this.width;
+        }
+        if (this.hasHeight) {
+          return this.widthFromHeight;
+        }
+        return '';
```

Comment:
Since `widthFromHeight` is now string,

```javascript
if (this.hasHeight) {
  return this.widthFromHeight;
}
return '';
```

can be simplified to

```javascript
return this.widthFromHeight;
```